src/config/paths-to-module-name-mapper.ts

```diff
@@ -47,12 +46,21 @@ export const pathsToModuleNameMapper = (
 
         return `${enrichedPrefix}${enrichedTarget.replace(/\*/g, '$1')}`
       })
-      pattern = `^${escapeRegex(segments[0])}(.*)${escapeRegex(segments[1])}$`
+
+      if (removeJsExtension) {
+        const patternWithJs = `^${escapeRegex(segments[0])}(.*)${escapeRegex(segments[1])}\\.js$`
+        jestMap[patternWithJs] = paths.length === 1 ? paths[0] : paths
+      }
+      const pattern = `^${escapeRegex(segments[0])}(.*)${escapeRegex(segments[1])}$`
       jestMap[pattern] = paths.length === 1 ? paths[0] : paths
     } else {
       logger.warn(interpolate(Errors.NotMappingMultiStarPath, { path: fromPath }))
     }
   }
 
+  if (removeJsExtension) {
```

Comment:
- Can we somehow make this block of codes to be at the `if (removeJsExtension)` above?
- Can we not introduce an extra regex here? Would it be an idea to omit the file extension on the final result of mapping?